crates/sargon/src/gateway_api/models/types/response/state/entity/details/details/item.rs

```diff
@@ -0,0 +1,251 @@
+            ComponentEntityRoleAssignmentEntry::new(
+                RoleKey::main_depositor(),
+                ComponentEntityRoleAssignmentEntryAssignment::sample_explicit_allow_all(
+                ),
+            ),
+            ComponentEntityRoleAssignmentEntry::new(
+                RoleKey::main_withdrawer(),
+                ComponentEntityRoleAssignmentEntryAssignment::sample_explicit_deny_all(),
+            ),
+        ]);
+        let explicit__depositor_deny_withdrawer_allow = Assignments::new(explicit.clone(), vec![
+            ComponentEntityRoleAssignmentEntry::new(
+                RoleKey::main_depositor(),
+                ComponentEntityRoleAssignmentEntryAssignment::sample_explicit_deny_all(),
+            ),
+            ComponentEntityRoleAssignmentEntry::new(
+                RoleKey::main_withdrawer(),
+                ComponentEntityRoleAssignmentEntryAssignment::sample_explicit_allow_all(
+                ),
+            ),
+        ]);
+        let owner__deny_all =
+            Assignments::new(
+                owner_deny_all.clone(),
+                vec![
+                    ComponentEntityRoleAssignmentEntry::sample_depositor_owner_allow_all(),
+                    ComponentEntityRoleAssignmentEntry::sample_withdrawer_owner_allow_all(),
+        ],
+            );
+
+        // Define a few assignments combinations that can be transferred
+        let explicit__allow_all = Assignments::sample_allow_all();
+        let owner__allow_all =
+            Assignments::new(owner_allow_all.clone(), vec![
+                ComponentEntityRoleAssignmentEntry::sample_depositor_owner_allow_all(),
+                ComponentEntityRoleAssignmentEntry::sample_withdrawer_owner_allow_all(),
+            ]);
+
+        // Define the two outcomes
+        let cannot_be_transferred = vec![
+            explicit__empty,
+            explicit__only_depositor,
+            explicit__depositor_allow_withdrawer_deny,
+            explicit__depositor_deny_withdrawer_allow,
+            owner__deny_all,
+        ];
+        let can_be_transferred = vec![explicit__allow_all, owner__allow_all];
+
+        // Test the combinations for FungibleResource
+        for assignments in cannot_be_transferred.clone() {
+            let sut = SUT::FungibleResource(
+                StateEntityDetailsResponseFungibleResourceDetails::new(
+                    assignments.clone(),
+                ),
+            );
+            assert!(!sut.can_be_transferred());
+        }
+        for assignments in can_be_transferred.clone() {
+            let sut = SUT::FungibleResource(
+                StateEntityDetailsResponseFungibleResourceDetails::new(
+                    assignments.clone(),
+                ),
+            );
+            assert!(sut.can_be_transferred());
+        }
+
+        // Test the combinations for NonFungibleResource
+        for assignments in cannot_be_transferred.clone() {
```

Comment:
have you run clippy fix? I think it will say that `clone()` is not needed, if you use `.iter()`